x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -609,6 +609,29 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
 
     const newPolicy = (await this.get(soClient, id)) as PackagePolicy;
 
+    // if we have moved to an input package we need to create the index templates
+    // for the package policy as input packages create index templates per package policy
+    if (pkgInfo && pkgInfo.type === 'input') {
```

Comment:
Maybe a small enhancement we can do here is to do the check only if the version is updated, what do you think?
```suggestion
    if (pkgInfo && pkgInfo.type === 'input' && oldPackagePolicy.package && oldPackagePolicy.package?.version !== pkgInfo.version) {
```